x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Attribute.java

```diff
@@ -32,6 +32,55 @@
  * The rest are not as they are not part of the projection and thus are not part of the derived table.
  */
 public abstract class Attribute extends NamedExpression {
+    /**
+     * A wrapper class where equality of the contained attribute ignores the {@link Attribute#id()}. Useful when we want to create new
+     * attributes and want to avoid duplicates - we can create a set of {@link NonSemanticAttribute}s.
+     */
+    public class NonSemanticAttribute {
```

Comment:
Is it intentionally not static? Also could it be a record to simplify construction and have a free toString while still having custom equals and hash code?

Reply:
Thanks, I think I wanted it to be non-static and then changed my mind :D turned into a record.